src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/DefaultCache.java

```diff
@@ -116,224 +97,99 @@ public DefaultCacheEntry get(
-        HttpCacheProxyCacheableRequest initialRequest)
-    {
-        PendingInitialRequests pendingInitialRequests = pendingInitialRequestsMap.get(initialRequest.requestHash());
-        pendingInitialRequests.subscribe(initialRequest);
-    }
-
-    public void createPendingInitialRequests(
-        HttpCacheProxyCacheableRequest initialRequest)
-    {
-        pendingInitialRequestsMap.put(initialRequest.requestHash(), new PendingInitialRequests(initialRequest));
-    }
-
-    public void removePendingInitialRequest(
-        HttpCacheProxyCacheableRequest request)
-    {
-        PendingInitialRequests pendingInitialRequests = pendingInitialRequestsMap.get(request.requestHash());
-        if (pendingInitialRequests != null)
-        {
-            pendingInitialRequests.removeSubscriber(request);
-            if(pendingInitialRequests.numberOfSubscribers() == 0)
-            {
-                pendingInitialRequestsMap.remove(request.requestHash());
-            }
-        }
-    }
-
-    public void serveNextPendingInitialRequest(HttpCacheProxyCacheableRequest request)
-    {
-        removePendingInitialRequest(request);
-        sendPendingInitialRequests(request.requestHash());
-    }
 
     public void purge(
         int requestHash)
     {
         DefaultCacheEntry cacheEntry = this.cachedEntries.remove(requestHash);
+        entryCount.accept(-1);
         if (cacheEntry != null)
         {
             cacheEntry.purge();
         }
     }
 
-    public boolean hasPendingInitialRequests(
-        int requestHash)
-    {
-        return pendingInitialRequestsMap.containsKey(requestHash);
-    }
-
-
-    public void purgeEntriesForNonPendingRequests()
-    {
-        cachedEntries.getCachedEntries().forEach((i, cacheEntry)  ->
-        {
-             if (!hasPendingInitialRequests(cacheEntry.requestHash()))
-             {
-                 this.purge(cacheEntry.requestHash());
-             }
-        });
-    }
 
     public boolean isUpdatedByEtagToRetry(
-        HttpCacheProxyCacheableRequest request,
+        ListFW<HttpHeaderFW> requestHeaders,
+        String ifNoneMatch,
         DefaultCacheEntry cacheEntry)
     {
-        ListFW<HttpHeaderFW> requestHeaders = request.getRequestHeaders(requestHeadersRO);
         ListFW<HttpHeaderFW> responseHeaders = cacheEntry.getCachedResponseHeaders();
         if (isPreferWait(requestHeaders)
             && !isPreferenceApplied(responseHeaders))
         {
             String status = HttpHeadersUtil.getHeader(responseHeaders, HttpHeaders.STATUS);
-            String etag = request.etag();
             String newEtag = cacheEntry.etag();
             assert status != null;
 
-            if (etag != null && newEtag != null)
+            if (ifNoneMatch != null && newEtag != null)
             {
-                return !(status.equals(HttpStatus.OK_200) && etag.equals(newEtag));
+                return !(status.equals(HttpStatus.OK_200) && ifNoneMatch.equals(newEtag));
```

Comment:
In practice `if-none-match` will likely only contain a single `etag` value, but by spec it is permitted to present more than one `etag` in the value of the `if-none-match` header.

Reply:
Done.